frontend/src/pages/destructionlist/detail/DestructionListDetail.tsx

```diff
@@ -169,25 +150,58 @@ export async function destructionListUpdateAction({
  * React Router loader.
  */
 export const destructionListDetailLoader = loginRequired(
-  async ({ request, params }: ActionFunctionArgs) => {
-    if (typeof params.uuid === "undefined") return {};
-
+  async ({
+    request,
+    params,
+  }: ActionFunctionArgs): Promise<DestructionListDetailContext> => {
+    const uuid = params.uuid as string;
+    const storageKey = `destruction-list-detail-${uuid}`;
+    const searchParams = Object.fromEntries(new URL(request.url).searchParams);
+
+    // Get reviewers, zaken and zaaktypen.
     const promises = [
-      getDestructionList(params.uuid),
-      getZakenData(request, {
-        not_in_destruction_list_except: String(params.uuid),
-      }),
+      getDestructionList(uuid as string),
       listReviewers(),
+      /*
+       Intercept and ignore 404 due to the following scenario cause by shared `page` parameter:
+
+       - User navigates to destruction list with 1 page of items.
+       - Users click edit button
+       - User navigates to page 2
+       - zaken API with param `in_destruction_list` may return 404.
+      */
+      listZaken({ ...searchParams, in_destruction_list: uuid }).catch((e) => {
+        if (e.status === 404) {
+          return {
+            count: 0,
+            next: null,
+            previous: null,
+            results: [],
+          };
+        }
+      }),
+      listZaken({
+        ...searchParams,
+        not_in_destruction_list_except: uuid,
+      }),
+      getZaakSelection(storageKey),
```

Comment:
why do we retrieve the selection in the loader instead of when the user starts the editing mode of the zaken?

Reply:
I'd say move as much I/O as possible to the loader, in this case it's a bit debatable though as we don't use the selection when not editing. I chose this for pure practical reasons. By fetching the selection in the loader we get a ready2go prop in the component instead of a Promise which is much harder to deal with. Since under de hood we only use session storage (we don't make an API call for this) its not much of a performance issue.

Technically we could optimize this a bit further by only fetching the selection when the view is in edit mode (which can also be detected by the loader) by now.